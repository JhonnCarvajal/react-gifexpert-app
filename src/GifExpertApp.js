import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";
const GifExpertApp = () => {
  //const categories = ["One Punch", "Samurai X", "Dragon Ball"];
  const [categories, setCategories] = useState(["One Punch"]);
  /*
  const handleAdd = () => {
    //setCategories([...categories,'HunterXHunter']);// Una soluci[n BIEN!!!!
    setCategories((cats) => [...cats, "HunterXHunter"]); //El segundo parametro que se envia al useStae
    //La funcion modificadora, tiene un callback que recibe como primer argumento el estado anterior!!!!!
  };
  */
  return (
    <>
      <h2>Gif Expert App</h2>
      <AddCategory setCategories={setCategories} />
      <hr />
      {/* <button onClick={()=>handleAdd()}>Agregar</button> */}
      <ol>
        {categories.map((category) => {
          return <GifGrid key={category} category={category} />;
        })}
      </ol>
    </>
  );
};

export default GifExpertApp;
