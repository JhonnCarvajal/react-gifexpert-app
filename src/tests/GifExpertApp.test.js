import { shallow } from "enzyme"
import GifExpertApp from "../GifExpertApp"

describe('Pruebas en GifExpertApp',()=>{
    
    test('Debe mostrar el componente correctamente', () => {
        const wrapper= shallow(<GifExpertApp  />)
        expect(wrapper).toMatchSnapshot();
    })
    test('Debe de mostrar una liosta de categorias', () => {
        const categories = ['One Punch','Goku']
        const wrapper= shallow(<GifExpertApp defaultCategories={categories} />)
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })
    
    

})