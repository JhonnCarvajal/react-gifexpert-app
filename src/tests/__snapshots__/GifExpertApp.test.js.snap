// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
